ui/src/app/applications/components/application-summary/application-summary.tsx

```diff
@@ -369,8 +369,9 @@ export const ApplicationSummary = (props: {app: models.Application; updateApp: (
             edit: null
         });
     const [badgeType, setBadgeType] = React.useState('URL');
-    const badgeURL = `${location.protocol}//${location.host}/api/badge?name=${props.app.metadata.name}&revision=true`;
-    const appURL = `${location.protocol}//${location.host}/applications/${props.app.metadata.name}`;
+    const rootPath = location.pathname.substring(0, location.pathname.indexOf('/applications/'));
```

Comment:
The root path is available in [Context.baseHref](https://github.com/argoproj/argo-cd/blob/master/ui/src/app/shared/context.ts#L11):

```typescript
import {Consumer, Context} from '../../../shared/context';
....

Please use it instead of inferring rootPath from the opened URL 
const context = React.useContext(Context);
console.info(context.baseHref);
```